client/components/EditWine.js

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect, useState } from "react";
+import { connect } from "react-redux";
+import { updateThisWine, fetchSingleWine } from "../store/singleWine";
+
+const EditWine = (props) => {
+  const { singleWine } = props;
+
+  const [wineInfo, setWineInfo] = useState({
+    name: "",
+    region: "",
+    year: "",
+    type: "",
+    typeOfGrape: "",
+    tastingNotes: "",
+    price: "",
+    quantity: "",
+    imageUrl: "",
+  });
+
+  useEffect(() => {
+    props.fetchSingleWine(props.singleWine.id);
+    setWineInfo(singleWine);
+  }, []);
+
+  const handleChange = (event) => {
+    setWineInfo({ ...wineInfo, [event.target.name]: event.target.value });
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    props.updateThisWine({ ...singleWine, ...wineInfo });
+    setWineInfo({
+      name: "",
+      region: "",
+      year: "",
```

Comment:
I think in handle submit we also need to set year, quantity, and price to numbers